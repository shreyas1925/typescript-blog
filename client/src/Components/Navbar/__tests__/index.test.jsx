import NavBar from ".."
import {render,screen} from '@testing-library/react'
describe('Navbar',()=>{
    it('should render correctly without crashing',()=>{
        render(<NavBar/>)
        const navbarElement = screen.getByTestId('navbar')
        expect(navbarElement).toBeTruthy()
    })

    it('should render all list items correctly',()=>{
        render(<NavBar/>)
        const listItems = screen.getAllByRole('listitem')
        expect(listItems).toHaveLength(3)
    })
    it('should render logo heading h1 correctly',()=>{
        render(<NavBar/>)
        const headingItems = screen.getByRole('heading',{
            level:1
        })
        expect(headingItems).toBeInTheDocument()
    })
    it('should render logo heading h3 correctly',()=>{
        render(<NavBar/>)
        const headingItems = screen.getByRole('heading',{
            level:3
        })
        expect(headingItems).toBeInTheDocument()
    })
    
    it('should render correctly without crashing forming a snapshot',()=>{
        const {asFragment} = render(<NavBar/>)
        expect(asFragment()).toMatchSnapshot()   
    })  
    
})