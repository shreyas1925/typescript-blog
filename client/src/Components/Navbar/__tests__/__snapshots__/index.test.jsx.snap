// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar should render correctly without crashing forming a snapshot 1`] = `
<DocumentFragment>
  <header
    data-testid="navbar"
  >
    <div
      class="logo--header"
    >
      <h1>
        The Artifact
      </h1>
      <h3>
        Culture & Art Blog
      </h3>
    </div>
    <nav>
      <ul>
        <li>
          Blog
        </li>
        <li>
          About
        </li>
        <li>
          Contact
        </li>
      </ul>
    </nav>
  </header>
</DocumentFragment>
`;
